### PR TITLE
Revise the authentication test to better reflect its functionality

### PR DESCRIPTION
**Pull request by PGijsbers**
The current design reflects the exact naming used in the REST API. However, I personally find that somewhat confusing for several reasons:

 - a method (or module) with `test` in the name immediately makes me think of unit tests, which is not what this is
 - the name describes a use case (testing what you are authorized for), while it is really doing something else (getting current user information) which is incidentally also useful to see . It also does not actually ensure that the provided roles authorize you to do anything.

Also, I think we should consider (minimally) wrapping the calls with e.g., custom exceptions instead of just forwarding the JSON with an error response. This is an example of what that could look like.

As for the return types, I am not sure yet. In general, having types is much nicer to work with. However, I am not sure yet what the best way to do this is. Ideally we carve out the models from the REST API and share them between both this package and the REST API. However, the REST API models are quite convoluted with read/write models and are tied into SQLModel. So I am not yet convinced that simple duplication is not a better solution.

So this is a proposal to rename this particular end point (and I would also propose to rename it on the REST API itself at some point), with some design suggestions for the API as a whole. Would like you hear your thoughts.

### CODE REVIEW

**Comment by jsmatias** [comment]
Regarding the name, I agree that renaming `test_authorization` to `get_current_user` will better reflect its functionality both in the REST API and SDK.

Regarding the implementation of wrapping calls with custom exceptions, I think it's a great approach to consider. However, I'd like to prioritise completing all calls to endpoints in the REST API, along with documenting the SDK. Once we've completed it and possibly released a MVP on PyPI, we can implement improvements. We are not far from having all the calls completed and documented. Once we have it, people from our team (and hopefully from the AI community) will be able to start using the SDK and we can possibly gather some feedbacks. 

For this particular case (the `test_authorization` method), I think it's okay to implement it now as you're suggesting. We can have it wrapped and return the user as a `NamedTuple`. It's a nice approach, and since it's not an AI asset, we have more freedom to change it.

**Comment by PGijsbers** [comment]
I agree that by and large the focus should be on wrapping the API calls. However, it is worth it to consider that these kind of modifications are also documentation themselves. If users get cryptic error messages, or don't even realize they have an error until much later down the line, they will be quick to throw in the towel (and/or contact us). 

Additionally, doing this later will mean breaking changes (functions which would return the error as JSON may now raise an exception instead). While we should clearly communicate that breaking changes are to be expected this early in the lifetime of the API, avoiding them is still nice.

Having some minimal improvements will not take us long, but likely saves both ourselves and our users time. Even in a short timeline. Just a thing to consider.

File: src/aiod/default/authorization_test.py [hunk 1]
```diff
@@ -1,18 +1,34 @@
+import http.client
+from collections import NamedTuple
+from typing import Sequence
```
Comment (jsmatias) [inline src/aiod/default/authorization_test.py:3]: `NamedTuple` to be used for inheritance in this case should come from `typing` instead `collections`. 
```suggestion
from typing import Sequence, NamedTuple
```

File: src/aiod/default/authorization_test.py [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
 
 from aiod.config.settings import api_base_url
 
 
-def test() -> requests.Response:
-    """
-    Send a test request to the authorization_test endpoint.
+class User(NamedTuple):
+    name: str
+    roles: Sequence[str, ...]
```
Comment (jsmatias) [inline src/aiod/default/authorization_test.py:11]: `Pylance` complains that `...` is not allowed in this context.
```suggestion
    roles: Sequence[str]
```

**Review by jsmatias — CHANGES_REQUESTED** [review]
Looks good to me. Just two comments on the syntax. For me, those lines are incorrect, but it doesn't seem to be a version conflict. Does it work for you?

**Comment by jsmatias** [comment]
> Looks good to me. Just two comments on the syntax. For me, those lines are incorrect, but it doesn't seem to be a version conflict. Does it work for you?

Sorry, just noticed that this branch is a work in progress. As I started working on the authentication process I can take this over and continue it.

**Review by jsmatias — APPROVED** [review]
I've updated the changes. Please have a look and we can merge this.

File: src/aiod/default/authorization_test.py (hunk 2 above)
Comment (PGijsbers) [reply src/aiod/default/authorization_test.py]: Yes I think this was changed from `Tuple[str, ...]`, must have missed the warning myself.

File: src/aiod/default/authorization_test.py (hunk 1 above)
Comment (PGijsbers) [reply src/aiod/default/authorization_test.py]: Yep, looks like my IDE imported the wrong version.

**Comment by PGijsbers** [comment]
Thanks! I just drafted this up quickly, I don't know if I even ran it (hence no tests). The mistakes you noticed were likely from a small refactor or IDE import.

**Pull request merged**